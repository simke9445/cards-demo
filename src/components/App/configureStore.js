import 'regenerator-runtime/runtime';
import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../../reducers';
import createSagaMiddleware from 'redux-saga';
import rootSaga from '../../sagas';

const sagaMiddleware = createSagaMiddleware();

export function configureStore( preloadedState ) {
	const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

	const store = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(
			applyMiddleware(
				sagaMiddleware,
			),
		),
	);

	sagaMiddleware.run(rootSaga);

	return store;
}
