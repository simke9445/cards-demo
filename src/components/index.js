export { default as App } from './App';
export { Card } from './Card';
export { Page } from './Page';
