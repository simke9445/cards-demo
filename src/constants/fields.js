export const fields = [
  'ukResident',
  'sex',
  'firstName',
  'lastName',
  'title',
];
