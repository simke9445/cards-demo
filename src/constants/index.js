export * from './locales';
export * from './fields';
