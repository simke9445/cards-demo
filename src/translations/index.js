import flatten from 'flat';
import { addLocaleData } from 'react-intl';

import enLocale from 'react-intl/locale-data/en';

import { LOCALES } from '../constants/locales';

import en from './en.yaml';

addLocaleData([...enLocale]);

export default {
	[LOCALES.EN]: flatten(en),
};
