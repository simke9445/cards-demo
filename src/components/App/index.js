import React from 'react';
import { connect } from 'react-redux';

import { BrowserRouter } from "react-router-dom";
import { IntlProvider } from "react-intl";
import { Provider as ReduxProvider } from "react-redux";

import { getLocale, getPage } from '../../selectors';
import translations from '../../translations';

import App from "./App";

import { configureStore } from "./configureStore";

const IntlApp = ( { locale, messages, page } ) => (
	<IntlProvider locale={locale} messages={messages}>
		<App page={page}/>
	</IntlProvider>
);

const ConnectedIntlApp = connect(
	state => {
		const locale = getLocale(state);
		const messages = translations[locale];
		const page = getPage(state);

		return { locale, messages, page };
	},
	null, null, { pure: false },
)(IntlApp);

const store = configureStore();

export default () => (
    <BrowserRouter>
        <ReduxProvider store={store}>
            <ConnectedIntlApp/>
        </ReduxProvider>
    </BrowserRouter>
)