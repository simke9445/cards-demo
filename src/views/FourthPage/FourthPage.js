import React from 'react';
import PropTypes from 'prop-types';

import { Page, Card } from '../../components';

const styles = {
  pre: {
    textAlign: 'start',
  },
};

const selections = ['title'];

const FourthPage = ({ pageIndex, onPrevious, formValue, intl }) => {
  const parsedFormValue = Object.keys(formValue)
    .filter(key => selections.indexOf(key) !== -1)
    .map(key => ({ [key]: intl.formatMessage({ id: formValue[key] }) }))
    .reduce((acc, curr) => ({ ...acc, ...curr }), formValue);

  return (
    <Page
      hasPrevious={true}
      onPrevious={onPrevious}
      pageIndex={pageIndex}
    >
      <Card>
        <pre style={styles.pre}><code>{JSON.stringify(parsedFormValue, null, 2)}</code></pre>
      </Card>
    </Page>
  );
}

FourthPage.propTypes = {
  formValue: PropTypes.object.isRequired,
};

export default FourthPage;