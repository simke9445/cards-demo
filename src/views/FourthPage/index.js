import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import { getFormValues } from 'redux-form';
import { compose } from 'recompose';

import { FourthPage } from './FourthPage';
import { getPage } from '../../selectors';
import { setPage } from '../../actions';

const EnhancedFourthPage = compose(
  connect(
    state => ({
      formValue: getFormValues('cards')(state),
      pageIndex: getPage(state),
    }),
    dispatch => ({
      onPrevious: (index) => dispatch(setPage({ index })),
    }),
    (stateProps, dispatchProps) => ({
      ...stateProps,
      ...dispatchProps,
      onPrevious: () => dispatchProps.onPrevious(stateProps.pageIndex - 1),
    }),
  ),
  injectIntl,
)(FourthPage);

export default EnhancedFourthPage;

