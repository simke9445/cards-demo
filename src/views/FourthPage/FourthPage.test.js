import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { default as ConnectedFourthPage } from './index'; 
import translations from '../../translations';
import configureStore from 'redux-mock-store';
import { LOCALES } from '../../constants';
import { Card, Page } from '../../components';
import FourthPage from './FourthPage';

it('FourthPage renders without crashing', () => {
  const formValue = { test };
  const wrapper = shallow(
    <IntlProvider messages={translations[LOCALES.EN]} locale={LOCALES.EN}>
      <FourthPage formValue={formValue}/>        
    </IntlProvider>
  );

  expect(wrapper).toMatchSnapshot();
})

it('ConnectedFourthPage renders without crashing', () => {
  const values = { test };
  const initialState = {
    user: { locale: LOCALES.EN },
    page: { index: 4 },
    form: {
      cards: {
        values,
      },
    },
  };
  const store = configureStore()(initialState);
  const wrapper = mount(
    <Provider store={store}>
      <IntlProvider messages={translations[LOCALES.EN]} locale={LOCALES.EN}>
        <ConnectedFourthPage/>        
      </IntlProvider>
    </Provider>
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(FourthPage).prop('formValue')).toEqual(values);  
  expect(wrapper.find(Page).length).toEqual(1);
  expect(wrapper.find(Card).length).toEqual(1);
  expect(wrapper.find(Page).prop('pageIndex')).toEqual(4);
  expect(wrapper.find(Page).prop('hasPrevious')).toEqual(true);
  expect(wrapper.find(Page).prop('hasNext')).toBeUndefined();
  expect(wrapper.find(Card).prop('hasNavigation')).toBeUndefined();
  expect(wrapper.find('div.half-circle').length).toEqual(1);
  expect(wrapper.find('div.half-circle-rotated').length).toEqual(1);
  expect(wrapper.find(Card).prop('hasTooltip')).toBeUndefined();
});
