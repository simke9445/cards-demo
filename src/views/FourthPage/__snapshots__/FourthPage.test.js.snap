// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedFourthPage renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <IntlProvider
    locale="en"
    messages={
      Object {
        "card.footer": "Why do you need this info?",
        "card.header": "Sign up",
        "page.title": "Moment",
        "page1.content": "Please tell us your name...",
        "page1.placeholder.firstName": "First Name",
        "page1.placeholder.lastName": "Last Name",
        "page1.select.default": "Choose your title",
        "page1.select.option1": "Mr",
        "page1.select.option2": "Mrs",
        "page1.select.option3": "Miss",
        "page1.select.option4": "Ms",
        "page1.select.option5": "Doctor",
        "page1.tooltip": "We need this information to proceed. Required fields are displayed in red.",
        "page2.button.female": "Female",
        "page2.button.male": "Male",
        "page2.content": "Please tell us your gender...",
        "page3.button.no": "No",
        "page3.button.yes": "Yes",
        "page3.content": "Are you a UK resident?",
        "page3.tooltip": "We are doing approximations based on your location.",
      }
    }
  >
    <Connect(FourthPage)>
      <FourthPage
        formValue={
          Object {
            "test": [Function],
          }
        }
        onPrevious={[Function]}
        pageIndex={4}
      >
        <FourthPage
          formValue={
            Object {
              "test": [Function],
            }
          }
          intl={
            Object {
              "formatMessage": [Function],
            }
          }
          onPrevious={[Function]}
          pageIndex={4}
        >
          <Page
            hasPrevious={true}
            onPrevious={[Function]}
            pageIndex={4}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F5F5F5",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": "100%",
                  "justifyContent": "flex-start",
                }
              }
            >
              <div
                className="pt-callout"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "boxShadow": "rgba(0, 0, 0, 0.15) 0px 3px 14px",
                    "display": "flex",
                    "height": "7.5%",
                    "justifyContent": "space-between",
                    "width": "90%",
                  }
                }
              >
                <h3
                  style={
                    Object {
                      "color": "#B2B6B9",
                      "paddingLeft": "14px",
                      "textAlign": "left",
                      "width": "20%",
                    }
                  }
                >
                  <FormattedMessage
                    id="page.title"
                    values={Object {}}
                  >
                    <span>
                      Moment
                    </span>
                  </FormattedMessage>
                </h3>
                <div
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                >
                  <div
                    className="pt-progress-bar pt-no-stripes"
                  >
                    <div
                      className="pt-progress-meter"
                      style={
                        Object {
                          "backgroundColor": "#A06FEF",
                          "display": "flex",
                          "width": "100%",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  style={
                    Object {
                      "visibility": "hidden",
                      "width": "20%",
                    }
                  }
                />
              </div>
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flex": "1",
                    "justifyContent": "space-between",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="half-circle"
                  onClick={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "cursor": "pointer",
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Blueprint.Icon
                    iconName="chevron-left"
                    style={
                      Object {
                        "color": "white",
                        "fontSize": "3em",
                        "transform": "translate(-10px)",
                      }
                    }
                  >
                    <span
                      className="pt-icon-standard pt-icon-chevron-left"
                      style={
                        Object {
                          "color": "white",
                          "fontSize": "3em",
                          "transform": "translate(-10px)",
                        }
                      }
                    />
                  </Blueprint.Icon>
                </div>
                <Card>
                  <div
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flex": "1",
                        "flexDirection": "column",
                        "height": "100%",
                        "justifyContent": "center",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      className="pt-card pt-elevation-2"
                      style={
                        Object {
                          "boxShadow": "rgba(0, 0, 0, 0.14) 0px 0em 2em 2px, rgb(242, 242, 242) 0px 0.75em 0px -0.25em, rgba(0, 0, 0, 0.09) 0px 1.5em 0.1875em -0.5em, rgb(229, 229, 229) 0px 1em 0px -0.5em, rgba(0, 0, 0, 0.1) 0px 1em 0.1875em -0.5em",
                          "display": "flex",
                          "flexDirection": "column",
                          "height": "500px",
                          "padding": "0",
                          "width": "340px",
                          "wordWrap": "break-word",
                        }
                      }
                    >
                      <div
                        className="pt-callout"
                        style={
                          Object {
                            "backgroundColor": "#A06FEF",
                          }
                        }
                      >
                        <h5
                          style={
                            Object {
                              "color": "white",
                            }
                          }
                        >
                          <FormattedMessage
                            id="card.header"
                            values={Object {}}
                          >
                            <span>
                              Sign up
                            </span>
                          </FormattedMessage>
                        </h5>
                      </div>
                      <div
                        style={
                          Object {
                            "display": "flex",
                            "flex": 1,
                            "flexDirection": "column",
                            "justifyContent": "center",
                            "padding": "24px",
                          }
                        }
                      >
                        <pre
                          style={
                            Object {
                              "textAlign": "start",
                            }
                          }
                        >
                          <code>
                            {}
                          </code>
                        </pre>
                      </div>
                      <div
                        className="pt-callout"
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "white",
                            "borderTop": "1px solid #DCDCDC",
                            "display": "flex",
                            "height": "10%",
                            "justifyContent": "center",
                          }
                        }
                      />
                    </div>
                  </div>
                </Card>
                <div
                  className="half-circle-rotated"
                  onClick={null}
                  style={
                    Object {
                      "alignItems": "center",
                      "cursor": "pointer",
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Blueprint.Icon
                    iconName="chevron-left"
                    style={
                      Object {
                        "color": "white",
                        "fontSize": "3em",
                        "transform": "translate(-10px)",
                      }
                    }
                  >
                    <span
                      className="pt-icon-standard pt-icon-chevron-left"
                      style={
                        Object {
                          "color": "white",
                          "fontSize": "3em",
                          "transform": "translate(-10px)",
                        }
                      }
                    />
                  </Blueprint.Icon>
                </div>
              </div>
            </div>
          </Page>
        </FourthPage>
      </FourthPage>
    </Connect(FourthPage)>
  </IntlProvider>
</Provider>
`;

exports[`FourthPage renders without crashing 1`] = `
<FourthPage
  formValue={
    Object {
      "test": [Function],
    }
  }
/>
`;
